data/crac-creation/crac-creator-csa-profiles/src/main/java/com/farao_community/farao/data/crac_creation/creator/csa_profile/crac_creator/contingency/CsaProfileContingencyCreator.java

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright (c) 2023, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+package com.farao_community.farao.data.crac_creation.creator.csa_profile.crac_creator.contingency;
+
+import com.farao_community.farao.data.crac_api.Crac;
+import com.farao_community.farao.data.crac_creation.creator.api.ImportStatus;
+import com.farao_community.farao.data.crac_creation.creator.csa_profile.crac_creator.CsaProfileConstants;
+import com.farao_community.farao.data.crac_creation.creator.csa_profile.crac_creator.CsaProfileCracCreationContext;
+import com.powsybl.iidm.network.Network;
+import com.powsybl.triplestore.api.PropertyBag;
+import com.powsybl.triplestore.api.PropertyBags;
+
+import java.util.HashSet;
+import java.util.Set;
+
+/**
+ * @author Jean-Pierre Arnould {@literal <jean-pierre.arnould at rte-france.com>}
+ */
+public class CsaProfileContingencyCreator {
+
+    private final Crac crac;
+
+    private final Network network;
+
+    private final PropertyBags contingenciesPropertyBags;
+
+    private final PropertyBags contingencyEquipmentsPropertyBags;
+
+    private Set<CsaProfileContingencyCreationContext> csaProfileContingencyCreationContexts;
+    private CsaProfileCracCreationContext cracCreationContext;
+
+    public CsaProfileContingencyCreator(Crac crac, Network network, PropertyBags contingenciesPropertyBags, PropertyBags contingencyEquipmentsPropertyBags, CsaProfileCracCreationContext cracCreationContext) {
+        this.crac = crac;
+        this.network = network;
+        this.contingenciesPropertyBags = contingenciesPropertyBags;
+        this.contingencyEquipmentsPropertyBags = contingencyEquipmentsPropertyBags;
+        this.cracCreationContext = cracCreationContext;
+    }
+
+    public void createAndAddContingencies() {
+        this.csaProfileContingencyCreationContexts = new HashSet<>();
+
+        for (PropertyBag contingencyPropertyBag : contingenciesPropertyBags) {
+            String contingencyId = contingencyPropertyBag.getId(CsaProfileConstants.REQUEST_CONTINGENCIES_RDFID);
+            String contingencyName = contingencyPropertyBag.get(CsaProfileConstants.REQUEST_CONTINGENCIES_NAME).concat(contingencyPropertyBag.get(CsaProfileConstants.REQUEST_CONTINGENCIES_EQUIPMENT_OPERATOR));
```

Comment:
I think it would be nice to have an EIC<->TSO name conversion table, because EIC is not very readable. Maybe we won't be able to map all EICodes, in that case if we read an EIC that is not mapped, we can use it as is. 
Here is the list of known TSOs with their EICodes : 
https://www.entsoe.eu/data/energy-identification-codes-eic/eic-approved-codes/ (type "System Operator" in "FUNCTION" column, and maybe start with only TSOs from CORE, IN, and SWE regions)